server/server_test.go

```diff
@@ -222,6 +222,31 @@ func TestOAuth2CodeFlow(t *testing.T) {
 			},
 		},
 		{
+			name: "verify at_hash",
+			handleToken: func(ctx context.Context, p *oidc.Provider, config *oauth2.Config, token *oauth2.Token) error {
+				rawIDToken, ok := token.Extra("id_token").(string)
+				if !ok {
+					return fmt.Errorf("no id token found")
+				}
+				idToken, err := p.Verifier().Verify(ctx, rawIDToken)
+				if err != nil {
+					return fmt.Errorf("failed to verify id token: %v", err)
+				}
+
+				var claims struct {
+					AtHash string `json:"at_hash"`
+				}
+				if err := idToken.Claims(&claims); err != nil {
+					return fmt.Errorf("failed to decode raw claims: %v", err)
+				}
+				if claims.AtHash == "" {
```

Comment:
at_hash is optional and there are lots of times you'd verify the id_token without having the access token. Don't know what the best way to introduce this to go-oidc is but I've opened https://github.com/coreos/go-oidc/issues/126 for good measure.